lib/sdam/topology_description.js

```diff
@@ -163,7 +163,7 @@ class TopologyDescription {
     }
 
     if (topologyType === TopologyType.Unknown) {
-      if (serverType === ServerType.Standalone) {
+      if (serverType === ServerType.Standalone && this.servers.size !== 1) {
```

Comment:
Why was it necessary to add this check against `this.servers.size`?

Reply:
My understanding of this is that we got back `Standalone` from the server, but the config had multiple servers, so it's not truly `Standalone` we don't process the `serverType` and get rid of this address. Although I'm not quite sure, this was ported over from 3.6 when Matt helped me with making the changes. I've tried testing combinations around this, it all seems to fail.

Reply:
Here's a test that came up failing referring to what I'm trying to explain: https://github.com/mongodb/specifications/blob/0cf00e6ec2ebc1f8959d04c674623431b58198f1/source/server-discovery-and-monitoring/tests/single/standalone_removed.json

Reply:
I think this is a translation of the [updateUnknownWithStandalone](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#updateunknownwithstandalone) guidance in the SDAM spec.

Reply:
OK to resolve @emadum ?

Reply:
Yep, thanks!